### PR TITLE
Add missing androidx support where appropriate

### PR DESCRIPTION
**Pull request by ZacSweers**
This adds analogous androidx classes alongside existing support library classes. This also adds an `isAndroidXCompatible` option to state and uses it for appropriate `NullnessFixes`.

### CODE REVIEW

File: core/src/main/java/com/google/errorprone/bugpatterns/nullness/NullnessFixes.java [hunk 1]
```diff
@@ -44,7 +45,7 @@ private static String getQualifiedName(VisitorState state, SuggestedFix.Builder
     Symbol sym = FindIdentifiers.findIdent("Nullable", state, KindSelector.VAL_TYP);
     String defaultType =
         state.isAndroidCompatible()
-            ? "android.support.annotation.Nullable"
+            ? (state.isAndroidXCompatible() ? "androidx.annotation.Nullable" : "android.support.annotation.Nullable" )
```
Comment (ronshapiro) [inline core/src/main/java/com/google/errorprone/bugpatterns/nullness/NullnessFixes.java]: I'm not sure this is an intended use of these types of flags. The ErrorProne team would be able to state authoritatively, but perhaps switching this to a check for if the type is available in the VisitorState would be more appropriate?

@awturner

Comment (ZacSweers) [reply core/src/main/java/com/google/errorprone/bugpatterns/nullness/NullnessFixes.java]: An alternative would just be to accept a "nullableAnnotation" option to the build itself, similar to the way the android option works.

**Comment by ZacSweers** [comment]
I'm unfortunately unable to properly test this, as master test sources seem to be largely broken?

image

image

image

File: core/src/main/java/com/google/errorprone/bugpatterns/nullness/NullnessFixes.java (hunk 1 above)
Comment (eaftan) [reply core/src/main/java/com/google/errorprone/bugpatterns/nullness/NullnessFixes.java]: I like the idea of passing in a flag with the fully qualified name of the nullableAnnotation to use.  I would not do it the way we do for -XDandroidCompatible though, I would use the Error Prong flags functionality: https://errorprone.info/docs/flags#pass-additional-info-to-bugcheckers

How about a flag like `-XepOpt:Nullness:NullableAnnotation`?

Comment (ZacSweers) [reply core/src/main/java/com/google/errorprone/bugpatterns/nullness/NullnessFixes.java]: That sounds good to me! I'll rework this PR to use that approach

**Comment by ZacSweers** [comment]
@eaftan updated with a new `-XepOpt:Nullness:NullableAnnotation` flag. Now nullness changes in this PR are:
- Default is `javax.annotation.Nullable`
- Custom fully qualified annotation as specified by the above flag
- Controlled by the `NullnessFixes` class
- The dagger `ProvidesNull` checker is updated to use `NullnessFixes` (CC @ronshapiro, this is the one I was telling you about). This also adds a new overload to accept a spacer suffix to the `makeFix` call, as this checker wants to always put the annotation on a new line.

Behavior changes worth mentioning:
- No longer tries to use an android annotation if the annotation error prone option is set. Must be set via the custom flag.
- `ProvidesNull` will use whatever `NullnessFixes` suggests, rather than always using `javax.annotation.Nullable`.

Let me know if/how you'd like to see tests for the integration and the API changes I made to support it. Similarly, I could also separate the NullnessFixes change to a separate PR since it's a wider scope.

**Pull request closed without merging**